File: 102. Binary Tree Level Order Traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    var result = [];
    if ( !root ) {
        return result;
    }

    var q = [];
    q.push(root);

    while ( q.length > 0 ) {
        var level = [];
        var length = q.length
        for ( var i = 0; i < length; i++ ) {
            var node = q.shift();
            level.push( node.val );
            if ( node.left ){
                q.push( node.left );
            }
            if ( node.right ){
                q.push( node.right );
            }
        }
        result.push( level );
    }

    return result;
};

File: 102. Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution(object):
    def levelOrder(self, root):
        result = []
        if root is None:
            return result
        q = []
        q.append(root)
        while len(q) > 0:
            length = len(q)
            row = []
            for i in xrange(length):
                node = q.pop(0)
                row.append(node.val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            result.append(row)
        return result
